README.md

```diff
@@ -76,317 +110,607 @@ cartridge --help
+* ``init.lua`` file which is the entry point for your application.
+* ``.git`` file necessary for a Git repository.
+* ``.gitignore`` file where you can specify the files to ignore.
+* ``env.lua`` file that sets common rock paths so that the application can be
+  started from any directory.
+* ``custom-role.lua`` file that is a placeholder for a custom (user-defined)
+  [cluster role](https://www.tarantool.io/en/doc/latest/book/cartridge/cartridge_dev/#cluster-roles).
+
+The entry point file (`init.lua`), among other things, loads the ``cartridge``
+module and calls its initialization function:
+
+```lua
+...
+local cartridge = require('cartridge')
+...
+cartridge.cfg({
+-- cartridge options example
+  workdir = '/var/lib/tarantool/app',
+  advertise_uri = 'localhost:3301',
+  cluster_cookie = 'super-cluster-cookie',
+  ...
+}, {
+-- box options example
+  memtx_memory = 1000000000,
+  ... })
+...
+```
+
+The `cartridge.cfg()` call renders the instance operable via the administrative
+console but does not call `box.cfg()` to configure instances.
+
+  **WARNING:** Calling the `box.cfg()` function is forbidden.
+  The cluster itself will do it for you when it is time to:
+
+  * bootstrap the current instance once you:
+
+    * run `cartridge.bootstrap()` via the administrative console, or
+    * click **Create** in the web interface;
+
+  * join the instance to an existing cluster once you:
+
+    * run `cartridge.join_server({uri = 'other_instance_uri'})` via the console, or
+    * click **Join** (an existing replica set) or **Create** (a new replica set)
+      in the web interface.
+
+Notice that you can specify a cookie for the cluster (`cluster_cookie` parameter)
+if you need to run several clusters in the same network. The cookie can be any
+string value.
+
+Now you can develop an application that will run on a single or multiple
+independent Tarantool instances (e.g. acting as a proxy to third-party databases)
+-- or will run in a cluster.
+
+  **NOTE:** If you plan to develop a cluster-aware application, first familiarize
+  yourself with the notion of
+  [cluster roles](https://www.tarantool.io/en/doc/latest/book/cartridge/cartridge_dev/#cluster-roles).
+
+### Building an application
+
+#### Building locally
+
+To build your application locally (for local testing), say this in any directory:
 
 ```sh
-cartridge build ./myapp
+cartridge build [<path>]
 ```
 
-Run instances locally:
+This command requires one argument -- the path to your application directory
+(i.e. to the build source). The default path is `.` (the current directory).
+
+This command runs:
+
+1. `cartridge.pre-build` (or [DEPRECATED] `.cartridge.pre`), if the
+   [pre-build file](#special-files) exists.
+   This builds the application in the `path` directory.
+2. `tarantoolctl rocks make`, if the [rockspec file](#special-files) exists.
+   This installs all Lua rocks to the `path` directory.
+
+During step 1, `cartridge` creates a temporary build directory (by default,
```

Comment:
No, the main idea of `build` command is that application is built exactly in it's directory.

Reply:
= we create a temp build dir only for `cartridge pack`?

Reply:
Yes